### PR TITLE
VEIOSS-505 remove self calculation that etmp already does and alter t…

### PR DESCRIPTION
**Pull request by DevAndyroo**
…he way we get return status' to use etmp obligations instead of get vat return, and reduce number of api calls

### CODE REVIEW

**Review by warrenskiwork1979 — DISMISSED** [review]

**Comment by DevAndyroo** [comment]
retest this please

File: app/uk/gov/hmrc/iossreturns/services/ReturnsService.scala [hunk 1]
```diff
@@ -102,24 +110,25 @@ class ReturnsService @Inject()(clock: Clock, vatReturnConnector: VatReturnConnec
...
       }
+    }
   }
 
   def getLastExclusionWithoutReversal(exclusions: List[EtmpExclusion]): Option[EtmpExclusion] = {
-
-      exclusions.maxByOption(_.effectiveDate).filterNot(_.exclusionReason == Reversal)
-
+    // Even though API is array ETMP only return single item
+    exclusions.headOption.filterNot(_.exclusionReason == Reversal)
```
Comment (pbyrne84) [inline app/uk/gov/hmrc/iossreturns/services/ReturnsService.scala:131]: in other places lastoption is used

https://github.com/hmrc/ioss-returns-frontend/blob/368f31ed19017f23a8ee7650bcec5b8e6eccbd0c/app/controllers/YourAccountController.scala#L51

Should probably agree on last of first, else potential for inconsistent weirdness?

Comment (DevAndyroo) [reply app/uk/gov/hmrc/iossreturns/services/ReturnsService.scala:131]: doesn't matter which, there will only ever be one

**Comment by DevAndyroo** [comment]
retest this please

**Review by warrenskiwork1979 — APPROVED** [review]

**Pull request merged**